cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/NameMappedDatasetFramework.java

```diff
@@ -151,7 +153,18 @@ public void deleteInstance(Id.DatasetInstance datasetInstanceId) throws DatasetM
                                           @Nullable Iterable<? extends Id> owners)
     throws DatasetManagementException, IOException {
     return super.getDataset(getMappedDatasetInstance(datasetInstanceId), arguments, classLoader,
-                               classLoaderProvider, owners);
+                            classLoaderProvider, owners);
+  }
+
+  @Nullable
+  @Override
+  public <T extends Dataset> T getDataset(Id.DatasetInstance datasetInstanceId, @Nullable Map<String, String> arguments,
```

Comment:
Shouldn't add it here. This class extends from `ForwardingDatasetFramework` and it should be added there since this method just call super.